src/swarm-topography/Swarm/Game/Scenario/Topography/Structure/Recognition/Log.hs

```diff
@@ -20,7 +20,7 @@ import Swarm.Game.Universe (Cosmic)
 
 renderSharedNames :: ConsolidatedRowReferences b a -> NonEmpty StructureName
 renderSharedNames =
-  NE.nub . NE.map (getName . originalDefinition . wholeStructure) . referencingRows
+  NE.nub . NE.map (name . originalItem . entityGrid . wholeStructure) . referencingRows
```

Comment:
Most likely the list is small (it's bounded by the number of user-defined structures/their sizes).  I was aware of this $O(N^2)$ issue but did not see an alternative in [`Data.List.NonEmpty`](https://hackage.haskell.org/package/base-4.20.0.1/docs/Data-List-NonEmpty.html#v:nub).   However, now that I look, I see [`Data.List.NonEmpty.Extra.nubOrd`](https://hackage.haskell.org/package/extra-1.8/docs/Data-List-NonEmpty-Extra.html#v:nubOrd), so I'll use that.

Reply:
Huh, `nubOrd` uses a custom red-black tree implementation.  I wonder why it doesn't just use `Set`.  Well, it doesn't matter.  Since the list we are nubbing is just a list of names, `nubOrd` should work well.

Reply:
Could the red-black implementation possibly be order-preserving?

Reply:
> I wonder why it doesn't just use `Set`.

I presume it is to minimize dependencies of the [`extra` package](https://hackage.haskell.org/package/extra).  Particularly, `extra` does not depend on [`containers`](https://hackage.haskell.org/package/containers-0.7/docs/Data-Set.html).

Incidentally, now that I've read the code I see that `nubOrd` is order-preserving whereas `nubSort` is not.  That fact would be good to note explicitly in the docstring (rather than indirectly by example).